flocker/node/agents/ebs.py

```diff
@@ -195,11 +317,74 @@ def resize_volume(self, blockdevice_id, size):
         pass
 
     # cloud_instance_id here too
-    def attach_volume(self, blockdevice_id, host):
-        pass
+    def attach_volume(self, blockdevice_id, attach_to):
+        """
+        Attach an EBS volume to given compute instance.
+
+        :param unicode blockdevice_id: EBS UUID for volume to be attached.
+        :param unicode attach_to: Instance id of AWS Compute instance to
+            attached the blockdevice to.
+
+        :raises UnknownVolume: If there does not exist a BlockDeviceVolume
+            corresponding to the input blockdevice_id.
+        :raises AlreadyAttachedVolume: If the input volume is already attached
+            to a device.
+        """
+        volume = self._get(blockdevice_id)
+        if volume.attached_to is not None:
+            raise AlreadyAttachedVolume(blockdevice_id)
+
+        with self.lock:
+            # begin lock scope
+
+            blockdevices = FilePath(b"/sys/block").children()
+            device = self._next_device(attach_to)
```

Comment:
Changed to:

            device = self._next_device(attach_to)

            if device is None:
                # TODO: Handle lack of free devices in ``/dev/sd[f-p]`` range
                # (see https://clusterhq.atlassian.net/browse/FLOC-1887).
                # No point in attempting an ``attach_volume``, so, return.
                return

            self.connection.attach_volume(blockdevice_id, attach_to, device)